### PR TITLE
feat: allow exist es

### PR DESCRIPTION
**Pull request by kierramarie**
### Description

Support for using an existing Elasticsearch db has been added to the DA. Variables and Outputs have been updated to reflect change.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by in-1911** [comment]
Other modules implementing this pattern (reference an existing instance) use CRN as a parameter. This actually makes it simpler to bind an existing instance because there is no need to know the ES instance resource group or region.

**Review by ocofaigh — DISMISSED** [review]
Agree - @kierramarie please be consistent with the other DAs and ask for an existing CRN

**Comment by shemau** [comment]
There are multiple issues that need resolution:

Indirectly related to this PR, but important is a test gap introduced in [1.15.0](https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/commit/a88e42315ad9a57e22169538ff4392df604a979b#diff-009e252987a2a3c3ea49bb0f9b3dc403769755965b5656e8b0a210d3499c0d6c). It appears that previously there were DA and DA upgrade tests. This was replaced with a single DA in schematics test.

This is important because module.elasticsearch changes to module.elasticsearch[0] under this PR. It is unclear if the upgrade path is tested, so it is unclear if a moved block is required or not.

The inputs were chosen because they are required for the data source for ibm_database. A CRN is not an option, but it may be possible to use a CRN to generate GUID and location/region via ibm_resource_instance; to get the name and resource group to pass to ibm_database. The GUID (from the CRN) should be unique and as such would not require scoping down the search by name/resource group on ibm_resource_instance data source.

Can we avoid elasticsearch in the variable/local/module names? This will help making the code more portable to other ICD modules. So maybe var.existing_instance_crn, local.use_existing_instance or local.use_existing_database, data.ibm_database.existing_instance and a new data.ibm_resource_instance.existing_instance.

Finally, are we confident that data.ibm_database_connection.existing_connection works? Do we know if that connection was created or should the connection information always be a new resource (rather than a data source)?

**Comment by kierramarie** [comment]
@shemau I have made the changes that were requested. I updated the variable names to say db_instance instead. I still wanted to distinguish between that and the existing_kms variables. Also, right now we do not have a way to verify that there is a connection. If it was made not using this DA it might not have one.

I also needed to add the provider bit to allow the resource_instance to work properly.

File: solutions/standard/main.tf
Comment (in-1911) [inline solutions/standard/main.tf:69]: You probably need to add `moved { }` block to avoid the instance being recreated on upgrade

Comment (shemau) [reply solutions/standard/main.tf:69]: Can we hold off adding a moved block until #244 is merged and the PR pipeline run here.

- sometimes terraform magic makes this sort of change just work
  - if it does, I would want to check it manually to verify the logs
  - if it does not, it would be valuable to catch it the new test

File: solutions/standard/variables.tf
Comment (shemau) [inline solutions/standard/variables.tf:44]: Can we make this more verbose. I think the version is a good template.

The CRN of an existing Databases for Elasticsearch instance. If no value is specified, a new instance is created.

File: solutions/standard/outputs.tf
Comment (shemau) [inline solutions/standard/outputs.tf:49]: This should be port (not hostname) in the true evaluation.

File: solutions/standard/provider.tf
Comment (shemau) [inline solutions/standard/provider.tf:11]: I do not understand the requirement for this. Two provider blocks using the same key. We should probably stay with one provider block until work is completed to support KMS is a different account.

Comment (ocofaigh) [reply solutions/standard/provider.tf:11]: The PR to add external KMS is merged - @kierramarie please rebase and address

**Comment by in-1911** [comment]
You need the `moved` block:
```
TestRunStandardUpgradeSolution 2024-08-02T15:57:43Z command.go:185:   # module.elasticsearch.module.elasticsearch.ibm_database.elasticsearch will be destroyed
TestRunStandardUpgradeSolution 2024-08-02T15:57:43Z command.go:185:   # (because module.elasticsearch.module.elasticsearch is not in configuration)
```

**Comment by shemau** [comment]
I think the moved statement should go in a separate file called moved.tf. Also the moved resource is just module.elasticsearch at the top level (which then goes to fscloud, in turn to the root module), but don't quote me on that.

```hcl
moved {
  from = module.elasticsearch
  to   = module.elasticsearch[0]
}
```

**Review by akocbek — APPROVED** [review]

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**
